src/main/java/wolox/training/controllers/UserController.java

```diff
@@ -0,0 +1,87 @@
+package wolox.training.controllers;
+
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+import wolox.training.models.Book;
+import wolox.training.models.User;
+import wolox.training.repositories.UserRepository;
+
+@Controller
+@RequestMapping(value = "/user")
+public class UserController {
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @GetMapping("/{id}")
+    public User findOne(@PathVariable Long id) {
+        return userRepository.findById(id)
+                .orElseThrow(RuntimeException::new);
+    }
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public User create(@RequestBody User user) {
+        return userRepository.save(user);
+    }
+
+    @DeleteMapping("/{id}")
+    public void delete(@PathVariable Long id) {
+        userRepository.findById(id)
+                .orElseThrow(RuntimeException::new);
+        userRepository.deleteById(id);
+    }
+
+    @PutMapping("/{id}")
+    public User updateUser(@RequestBody User user, @PathVariable Long id) {
+        if (user.getId() != id) {
+            throw new RuntimeException();
+        }
+        userRepository.findById(id)
+                .orElseThrow(RuntimeException::new);
+        return userRepository.save(user);
+    }
+
+
+    public void addBooks(Book book, Long idUser){
+        User userAux= findOne(idUser);
+        Boolean encontrado=false;
+        for (Book bookAux: userAux.getBooks()) {
+
+            if(bookAux.getId() == book.getId()){
+               encontrado = true;
```

Comment:
you can throw the exception directly when you find the match, there's no need to assign a variable an the throw if the variable is truthy